src/trajectory.jl

```diff
@@ -110,23 +146,23 @@ Ref: https://github.com/stan-dev/stan/blob/develop/src/stan/mcmc/hmc/nuts/base_n
 MultinomialTS(rng::AbstractRNG, z0::PhasePoint) = MultinomialTS(z0, zero(energy(z0)))
 
 """
-    SliceTS(s::SliceTS, H0::AbstractFloat, zcand::PhasePoint)
+    init_leafsampler(s::SliceTS, H0::AbstractFloat, zcand::PhasePoint)
```

Comment:
`leafsampler` is a vague term, I think. I prefer keeping the old type name based constructor here.

Reply:
Reverted.